src/Chainweb/Rosetta/Internal.hs

```diff
@@ -242,37 +244,52 @@ gasTransactionAcc accTyp txa@(TxAccumulator logs' _) ctx = combine logs'
   where
     combine (fundLog:someLog:restLogs) =
       case (getSomeTxId ctx) of
-        Nothing -> -- tx was unsuccessful
+        Nothing -> do -- tx was unsuccessful
+          assertSequentialTxIds (txId fundLog) (txId someLog)
           makeAcc restLogs
-          (makeOps FundTx fundLog)
-          [] -- no transfer logs
-          (makeOps GasPayment someLog)
-        Just tid   -- tx was successful
-          | tid /= (fst someLog) ->  -- tx didn't touch coin table
-            makeAcc restLogs
             (makeOps FundTx fundLog)
             [] -- no transfer logs
             (makeOps GasPayment someLog)
+        Just tid   -- tx was successful
+          | tid /= (txId someLog) -> do -- if tx didn't touch coin table
+            assertSequentialTxIds (txId fundLog) tid
+            assertSequentialTxIds tid (txId someLog)
+            makeAcc restLogs
+              (makeOps FundTx fundLog)
+              [] -- no transfer logs
+              (makeOps GasPayment someLog)
           | otherwise -> case restLogs of
-              gasLog:restLogs' -> makeAcc restLogs'
-                (makeOps FundTx fundLog)
-                (makeOps TransferOrCreateAcct someLog)
-                (makeOps GasPayment gasLog)
+              gasLog:restLogs' -> do  -- if tx DID touch coin table
+                assertSequentialTxIds (txId fundLog) (txId someLog)
+                assertSequentialTxIds (txId someLog) (txId gasLog)
+                makeAcc restLogs'
+                  (makeOps FundTx fundLog)
+                  (makeOps TransferOrCreateAcct someLog)
+                  (makeOps GasPayment gasLog)
               l -> listErr "No gas logs found after transfer logs" l
-    combine l = listErr "No fund and gas logs found" l
+    combine (f:[]) = listErr "Only fund logs found" f
+    combine [] = listErr "No logs found" ([] :: [(TxId, [AccountLog])])
 
     makeAcc restLogs fund transfer gas = pure $
       accumulatorFunction accTyp txa restLogs tx
       where
         tx = makeRosettaTx ctx $ indexedOperations $
              fund <> transfer <> gas
 
+    txId (tid,_) = tid
+
     makeOps ot (tid, als) =
       map (operation Successful ot tid) als
 
     listErr expectedMsg logs = Left $
       expectedMsg ++ ": Received logs list " ++ show logs
 
+    -- NOTE: no way to check fund's txId and prev txId seen are sequential
+    -- Assumption: Operations related to a transaction have sequential txIds.
+    assertSequentialTxIds tid1 tid2
+      | tid2 == succ tid1 = pure ()
+      | otherwise = Left $ "Expected sequential txId, found txId1="
+        ++ show tid1 ++ ", txId2=" ++ show tid2
```

Comment:
i don't think this is going to be true in general for someone's transactions. We enforce this by construction as well. It's probably unnecessary to test

Reply:
took it out in latest commit